### PR TITLE
resolves #1226 UrlButton: add an option to open the url in the same o…

### LINKED ISSUE #1226: [WEB_CONNECTOR] UrlButton: add an option to open the url in the same or another window
The urlButton allows to send a button with a url.
For the moment the choice to open the url in the url in the same window or in another one is mode in the client side.
Adding a parameter to the UrlButton indicating the client how the url should be opened from the server side will give more flexibility to the component.

### PR DESCRIPTION
**Pull request by bakic**
…r another window

### CODE REVIEW

**Comment by vsct-jburet** [comment]
@bakic does it means that the legacy behaviour is target:blank for the url ?

File: bot/connector-web/src/main/kotlin/WebBuilders.kt [hunk 1]
```diff
@@ -103,11 +103,13 @@ fun <T : Bus<T>> T.webButton(
 fun <T : Bus<T>> T.webUrlButton(
     title: CharSequence,
     url: String,
+    openInSameWindow: Boolean = false,
```
Comment (francoisno) [inline bot/connector-web/src/main/kotlin/WebBuilders.kt:106]: Adding the new param after `imageUrl` wouldn't be safer? (for callers, not naming the params)

File: bot/connector-web-model/src/main/kotlin/ai/tock/bot/connector/web/send/UrlButton.kt [hunk 2]
```diff
@@ -22,5 +22,6 @@ import com.fasterxml.jackson.annotation.JsonTypeName
 data class UrlButton(
     val title: String,
     val url: String,
+    val openInSameWindow: Boolean = false,
```
Comment (francoisno) [inline bot/connector-web-model/src/main/kotlin/ai/tock/bot/connector/web/send/UrlButton.kt:25]: Maybe `target` would be a better name than `window` and `targetBlank` or `targetSelf` better, to fit W3C attribute? (https://www.w3schools.com/tags/att_a_target.asp)

File: bot/connector-web/src/test/resources/card_with_buttons.json [hunk 3]
```diff
@@ -20,6 +20,7 @@
             "title": "title",
             "url": "http://www.sncf.com",
             "type": "web_url",
+            "openInSameWindow": "false",
```
Comment (francoisno) [inline bot/connector-web/src/test/resources/card_with_buttons.json:23]: `"openInSameWindow": false,` sans les quotes ? (pour être iso les autres fichiers de test)

**Review by francoisno — CHANGES_REQUESTED** [review]

File: bot/connector-web/src/main/kotlin/WebBuilders.kt (hunk 1 above)
Comment (bakic) [reply bot/connector-web/src/main/kotlin/WebBuilders.kt]: ok

File: bot/connector-web-model/src/main/kotlin/ai/tock/bot/connector/web/send/UrlButton.kt (hunk 2 above)
Comment (bakic) [reply bot/connector-web-model/src/main/kotlin/ai/tock/bot/connector/web/send/UrlButton.kt]: I changed the name of the attribute to target and added an enum that gathers the possible values of the target param to fit W3C values for the target attribute

File: bot/connector-web/src/test/resources/card_with_buttons.json (hunk 3 above)
Comment (bakic) [reply bot/connector-web/src/test/resources/card_with_buttons.json:23]: @francoisno please check the new commit

Comment (francoisno) [reply bot/connector-web/src/test/resources/card_with_buttons.json]: I just pushed a fix of test resources (2 `openInSameWindow` remained).
Please let me know if it's ok for you @bakic

**Review by francoisno — APPROVED** [review]

**Pull request merged**